vm/rust/src/lib.rs

```diff
@@ -122,39 +141,49 @@ pub extern "C" fn cairoVMCall(
         }
     }
 
+    let version_constants = get_versioned_constants(block_info.version);
+    let sierra_version_str = unsafe { CStr::from_ptr(sierra_version) }.to_str().unwrap();
+    let sierra_version =
+        SierraVersion::from_str(sierra_version_str).unwrap_or(SierraVersion::DEPRECATED);
+    let initial_gas: u64;
+    if sierra_version < SierraVersion::new(1, 7, 0) {
+        initial_gas = version_constants.infinite_gas_for_vm_mode();
+    } else {
+        initial_gas = version_constants.os_constants.validate_max_sierra_gas.0;
+    }
```

Comment:
Suggestion, initialize `initial_gas` directly. I know it might seem odd, but it's usually the style when writing rust code

```suggestion
    let initial_gas: u64 = if sierra_version < SierraVersion::new(1, 7, 0) {
        version_constants.infinite_gas_for_vm_mode();
    } else {
        version_constants.os_constants.validate_max_sierra_gas.0;
    }
```